### PR TITLE
AEA-1557 libp2p utils writebytes better error handling

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

libp2p utils writebytes better error handling

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/go/libp2p_node/libp2p_node.go [hunk 1]
```diff
@@ -157,8 +157,11 @@ func main() {
 		for envel := range agent.Queue() {
 			envelope := envel
 			logger.Info().Msgf("received envelope from agent: %s", envelope)
-			err := node.RouteEnvelope(envelope)
-			ignore(err)
+			go func() {
+				err := node.RouteEnvelope(envelope)
+				logger.Error().Msgf("Route envelope: %s", envelope)
+				ignore(err)
+			}()
```
[the remark this answers is not in the document]
Comment (solarw) [reply libs/go/libp2p_node/libp2p_node.go]: updated

File: examples/tac_deploy/tac-deployment.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ spec:
         kubernetes.io/os: linux
       containers:
       - name: tac-deploy-container
-        image: gcr.io/fetch-ai-sandbox/tac_deploy:0.0.7
+        image: gcr.io/fetch-ai-sandbox/tac_deploy:0.0.7-t2
```
[the remark this answers is not in the document]
Comment (solarw) [reply examples/tac_deploy/tac-deployment.yaml]: updated

File: examples/tac_deploy/build.sh [hunk 3]
```diff
@@ -4,13 +4,13 @@ set -e
 mkdir /data
 
 # setup the agent
-aea fetch fetchai/tac_controller:latest
+aea fetch --local fetchai/tac_controller:latest
```
[the remark this answers is not in the document]
Comment (solarw) [reply examples/tac_deploy/build.sh]: updated

File: libs/go/libp2p_node/utils/utils.go [hunk 4]
```diff
@@ -612,6 +613,14 @@ func FetchAIPublicKeyFromFetchAIPrivateKey(privateKey string) (string, error) {
 
 // WriteBytesConn send bytes to `conn`
 func WriteBytesConn(conn net.Conn, data []byte) error {
+	if len(data) > math.MaxInt32 {
+		logger.Error().Msg("data size too large")
+		return errors.New("data size too large")
```
Comment (marcofavorito) [inline libs/go/libp2p_node/utils/utils.go:618]: avoid repeating the same string

File: libs/go/libp2p_node/utils/utils.go [hunk 5]
```diff
@@ -612,6 +613,14 @@ func FetchAIPublicKeyFromFetchAIPrivateKey(privateKey string) (string, error) {
...
+	}
+	if len(data) == 0 {
+		logger.Error().Msg("No data to write")
+		return nil
```
Comment (marcofavorito) [inline libs/go/libp2p_node/utils/utils.go:622]: why `nil`?

**Review by marcofavorito — APPROVED** [review]

File: libs/go/libp2p_node/utils/utils.go (hunk 4 above)
Comment (solarw) [reply libs/go/libp2p_node/utils/utils.go:618]: one log, another error so it's ok

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**